### PR TITLE
cekit 4.10.0 (new formula)

### PR DESCRIPTION
**Pull request by gdams**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/c/cekit.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+class Cekit < Formula
+  include Language::Python::Virtualenv
+
+  desc "Container Evolution Kit"
+  homepage "https://cekit.io"
+  url "https://github.com/cekit/cekit/archive/refs/tags/4.10.0.tar.gz"
```
Comment (chenrui333) [inline Formula/c/cekit.rb:6]:
```suggestion
  url "https://files.pythonhosted.org/packages/c4/cc/a948fb7fc87bb974f55b37da024734166d4981a4f43958f1efc84d2b1e3d/cekit-4.10.0.tar.gz"
```

File: Formula/c/cekit.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  sha256 "e559014ed631e70375ab691799a09294465fba7bdf7846dd431301626578e7ba"
+  license "MIT"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
+  depends_on "python-click"
+  depends_on "python@3.11"
```
Comment (chenrui333) [inline Formula/c/cekit.rb:16]: does it support py3.12?

File: Formula/c/cekit.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  homepage "https://cekit.io"
+  url "https://github.com/cekit/cekit/archive/refs/tags/4.10.0.tar.gz"
+  sha256 "e559014ed631e70375ab691799a09294465fba7bdf7846dd431301626578e7ba"
+  license "MIT"
+
+  livecheck do
+    url :stable
+    strategy :github_latest
+  end
+
```
Comment (chenrui333) [inline Formula/c/cekit.rb:14]: if switch to pypi, this is not needed
```suggestion
```

File: Formula/c/cekit.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  def install
+    cd "cekit" do
+      virtualenv_install_with_resources
+
+      generate_completions_from_executable(bin/"cekit", shells: [:fish, :zsh], shell_parameter_format: :click)
+    end
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/cekit --version")
```
Comment (chenrui333) [inline Formula/c/cekit.rb:27]: need a more substantial test.

File: Formula/c/cekit.rb (hunk 1 above)
Comment (gdams) [reply Formula/c/cekit.rb:6]: Thanks, I'll apply

File: Formula/c/cekit.rb (hunk 2 above)
Comment (gdams) [reply Formula/c/cekit.rb:16]: yeah I'll update it

File: Formula/c/cekit.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+class Cekit < Formula
+  include Language::Python::Virtualenv
+
+  desc "Container Evolution Kit"
+  homepage "https://cekit.io"
+  url "https://files.pythonhosted.org/packages/c4/cc/a948fb7fc87bb974f55b37da024734166d4981a4f43958f1efc84d2b1e3d/cekit-4.10.0.tar.gz"
+  sha256 "83909f20463073aeb2d5f56295b7446c39ea4605fe8640ccfb521c618c5663e6"
+  license "MIT"
+
+  depends_on "python-click"
```
Comment (branchvincent) [inline Formula/c/cekit.rb:10]: this needs more than just click and pyyaml, no? `brew update-python-resources cekit` will automatically generate all the `resource`'s for you

```console
$ pipgrip --tree cekit
cekit (4.10.0)
├── click>=6.7 (8.1.7)
├── colorlog>=2.10.0 (6.8.2)
├── jinja2>=2.7 (3.1.3)
│   └── markupsafe>=2.0 (2.1.5)
├── odcs>=0.2.10 (0.8.0)
├── packaging>=19.0 (23.2)
├── pykwalify>=1.6.0 (1.8.0)
│   ├── docopt>=0.6.2 (0.6.2)
│   ├── python-dateutil>=2.8.0 (2.8.2)
│   │   └── six>=1.5 (1.16.0)
│   └── ruamel-yaml>=0.16.0 (0.18.5)
│       └── ruamel-yaml-clib>=0.2.7 (0.2.8)
└── pyyaml>=3.10 (6.0.1)

File: Formula/c/cekit.rb [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  depends_on "python@3.12"
+  depends_on "pyyaml"
+
+  def install
+    cd "cekit" do
+      virtualenv_install_with_resources
+
+      generate_completions_from_executable(bin/"cekit", shells: [:fish, :zsh], shell_parameter_format: :click)
+    end
```
Comment (branchvincent) [inline Formula/c/cekit.rb:19]:
```suggestion
    virtualenv_install_with_resources

    generate_completions_from_executable(bin/"cekit", shells: [:fish, :zsh], shell_parameter_format: :click)
```

**Comment by gdams** [comment]
@branchvincent this should be good to go now

File: Formula/c/cekit.rb [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+  include Language::Python::Virtualenv
+
+  desc "Container Evolution Kit"
+  homepage "https://cekit.io"
+  url "https://files.pythonhosted.org/packages/c4/cc/a948fb7fc87bb974f55b37da024734166d4981a4f43958f1efc84d2b1e3d/cekit-4.10.0.tar.gz"
+  sha256 "83909f20463073aeb2d5f56295b7446c39ea4605fe8640ccfb521c618c5663e6"
+  license "MIT"
+
+  depends_on "python-setuptools" => :build
+  depends_on "python-certifi"
```
Comment (branchvincent) [inline Formula/c/cekit.rb:11]: don't think this is needed:
```suggestion
```

**Review by branchv — APPROVED** [review]

**Comment by gdams** [comment]
@branchvincent would you mind approving again please and then we can merge this

**Review by SMillerDev — APPROVED** [review]

**Review by branchv — APPROVED** [review]

**Pull request merged**